src/careamics/models/lvae/layers.py

```diff
@@ -1333,666 +1332,4 @@ def forward(
-        }
-        return kl_dict
-
-    def process_p_params(
-        self, p_params: torch.Tensor, var_clip_max: float
-    ) -> Tuple[torch.Tensor, torch.Tensor, torch.distributions.normal.Normal]:
-        """
-        Process the input parameters to get the prior distribution p(z_i|z_{i+1}) (or p(z_L)).
-
-        Processing consists in:
-            - (optionally) 2D convolution on the input tensor to increase number of channels.
-            - split the resulting tensor into two chunks, the mean and the log-variance.
-            - (optionally) clip the log-variance to an upper threshold.
-            - define the normal distribution p(z) given the parameter tensors above.
-
-        Parameters
-        ----------
-        p_params: torch.Tensor
-            The input tensor to be processed.
-        var_clip_max: float
-            The maximum value reachable by the log-variance of the latent distribution.
-            Values exceeding this threshold are clipped.
-        """
-        if self.transform_p_params:
-            p_params = self.conv_in_p(p_params)
-        else:
-            assert p_params.size(1) == 2 * self.c_vars
-
-        # Define p(z)
-        p_mu, p_lv = p_params.chunk(2, dim=1)
-        if var_clip_max is not None:
-            p_lv = torch.clip(p_lv, max=var_clip_max)
-
-        p_mu = StableMean(p_mu)
-        p_lv = StableLogVar(p_lv, enable_stable=not self._use_naive_exponential)
-        p = Normal(p_mu.get(), p_lv.get_std())
-        return p_mu, p_lv, p
-
-    def process_q_params(
-        self, q_params: torch.Tensor, var_clip_max: float, allow_oddsizes: bool = False
-    ) -> Tuple[torch.Tensor, torch.Tensor, torch.distributions.normal.Normal]:
-        """
-        Process the input parameters to get the inference distribution q(z_i|z_{i+1}) (or q(z|x)).
-
-        Processing consists in:
-            - 2D convolution on the input tensor to increase number of channels.
-            - split the resulting tensor into two chunks, the mean and the log-variance.
-            - (optionally) clip the log-variance to an upper threshold.
-            - (optionally) crop the resulting tensors to ensure that the last spatial dimension is even.
-            - define the normal distribution q(z) given the parameter tensors above.
-
-        Parameters
-        ----------
-        p_params: torch.Tensor
-            The input tensor to be processed.
-        var_clip_max: float
-            The maximum value reachable by the log-variance of the latent distribution.
-            Values exceeding this threshold are clipped.
-        """
-        q_params = self.conv_in_q(q_params)
-
-        q_mu, q_lv = q_params.chunk(2, dim=1)
-        if var_clip_max is not None:
-            q_lv = torch.clip(q_lv, max=var_clip_max)
-
-        if q_mu.shape[-1] % 2 == 1 and allow_oddsizes is False:
-            q_mu = F.center_crop(q_mu, q_mu.shape[-1] - 1)
-            q_lv = F.center_crop(q_lv, q_lv.shape[-1] - 1)
```

Comment:
This will fail for 3D since `F.center_crop` only works in 3D. A solution is to replace it with `crop_to_img` defined somewhere in the codebase (likely in `lvae/utils`).